packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -352,12 +466,13 @@ export class ForkChoice implements IForkChoice {
     const blockRoot = this.config.getForkTypes(slot).BeaconBlock.hashTreeRoot(block);
     const blockRootHex = toHexString(blockRoot);
 
-    // Add proposer score boost if the block is timely
+    // Assign proposer score boost if the block is timely
     // before attesting interval = before 1st interval
+    const isBeforeAttestingInterval = blockDelaySec < this.config.SECONDS_PER_SLOT / INTERVALS_PER_SLOT;
+    const isTimely = this.fcStore.currentSlot === slot && isBeforeAttestingInterval;
```

Comment:
would be nice to implement `isBlockTimely()` to be used by e2e https://github.com/ChainSafe/lodestar/pull/6298/files#diff-a98ec83eae786124da6959cdd1681f4feb32256e0312d844bfb1abbeb8e04a68R1056
could be in the next PR through